File: LeetCode_Marathons/LCM6/Easy/Detect_Capital/a.go
func isMonotic(word string, isUpper bool) bool {
    for _, letter := range word {
        if isUpper && !('A' <= letter && letter <= 'Z') {
            return false
        }
        if !isUpper && !('a' <= letter && letter <= 'z') {
            return false
        }
    }

    return true
}

func detectCapitalUse(word string) bool {
    if 'a' <= word[0] && word[0] <= 'z' {
        return isMonotic(word[1:], false)
    }
    return isMonotic(word[1:], false) || isMonotic(word[1:], true)
}

File: LeetCode_Marathons/LCM6/Easy/Detect_Capital/a.java
class Solution {
    private boolean isMonotic(String word, boolean isUpper) {
        for (Character letter : word.toCharArray()) {
            if (isUpper && !('A' <= letter && letter <= 'Z')) {
                return false;
            }
            if (!isUpper && !('a' <= letter && letter <= 'z')) {
                return false;
            }
        }

        return true;
    }

    public boolean detectCapitalUse(String word) {
        if ('a' <= word.charAt(0) && word.charAt(0) <= 'z') {
            return isMonotic(word.substring(1), false);
        }
        return isMonotic(word.substring(1), false) || isMonotic(word.substring(1), true);
    }
}